src/libraries/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -792,21 +792,51 @@ internal static bool IsBidiControlCharacter(char ch)
         //
         // Strip Bidirectional control characters from this string
         //
-        internal static unsafe string StripBidiControlCharacter(char* strToClean, int start, int length)
+        internal static unsafe string StripBidiControlCharacters(char* strToClean, int length, string? backingString = null)
```

Comment:
As long as you're rewriting this, did you consider instead having it take a `ReadOnlySpan<char>` instead of strToClean + length?  That would remove the need for you to have the string-based overload, you wouldn't need to pin an input string, and you could eliminate most of the unsafe code being used.

Reply:
Changed it to use Span now